src/ray/object_manager/object_manager.cc

```diff
@@ -24,12 +29,41 @@ namespace object_manager_protocol = ray::object_manager::protocol;
 
 namespace ray {
 
+ObjectStoreRunner::ObjectStoreRunner(const ObjectManagerConfig &config) {
+  if (config.object_store_memory > 0) {
+    plasma_store_.reset(new plasma::PlasmaStoreRunner(
+        config.store_socket_name, config.object_store_memory, config.huge_pages,
+        config.plasma_directory, ""));
+    // Initialize object store.
+    store_thread_ = std::thread(&plasma::PlasmaStoreRunner::Start, plasma_store_.get());
+    // Sleep for sometime until the store is working. This can suppress some
+    // connection warnings.
+    std::this_thread::sleep_for(std::chrono::microseconds(500));
+  }
+}
+
+ObjectStoreRunner::~ObjectStoreRunner() {
+  if (plasma_store_ != nullptr) {
+    // Try to stop the eventloop gracefully with a timeout.
+    plasma_store_->Stop();
+    std::this_thread::sleep_for(std::chrono::microseconds(500));
+#ifdef _WIN32
+    TerminateThread(store_thread_.native_handle(), 0);
+#else
+    pthread_kill(store_thread_.native_handle(), SIGTERM);
```

Comment:
This isn't a good way to gracefully terminate threads in POSIX, and the Windows version is actually a forceful termination instead of a graceful one (which shouldn't be done). Let's figure out a way to send a message to its message loop so that it can be terminated properly before we merge; I can help you out.